heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -379,6 +399,101 @@ export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
     ];
   }
 
+  // Resolve all specified properties using Node resolution, and replace <rootDir> with the same rootDir
+  // that we provide to Jest. Resolve if we modified since paths containing <rootDir> should be absolute.
+  private static _getJsonPathMetadata(options: IJsonPathMetadataOptions): IJsonPathMetadata {
+    return {
+      customResolver: (configurationFilePath: string, value: string) => {
+        const configurationFileDir: string = path.dirname(configurationFilePath);
+        let packageDirMatches: RegExpExecArray | null;
+
+        // Compare with replaceRootDirInPath() from here:
+        // https://github.com/facebook/jest/blob/5f4dd187d89070d07617444186684c20d9213031/packages/jest-config/src/utils.ts#L58
+        if (value.startsWith(JestPlugin._rootDirToken)) {
+          // Example:  <rootDir>/path/to/file.js
+          const restOfPath: string = path.normalize('./' + value.substr(JestPlugin._rootDirToken.length));
+          value = path.resolve(options.buildFolder, restOfPath);
+        } else if (value.startsWith(JestPlugin._configDirToken)) {
+          // Example:  <configDir>/path/to/file.js
+          const restOfPath: string = path.normalize('./' + value.substr(JestPlugin._configDirToken.length));
+          value = path.resolve(configurationFileDir, restOfPath);
+        } else if ((packageDirMatches = JestPlugin._packageDirRegex.exec(value)) !== null) {
+          // Example:  <packageDir:@my/package>/path/to/file.js
+          const packageName: string | undefined = packageDirMatches.groups?.[JestPlugin._packageCaptureGroup];
+          if (!packageName) {
+            throw new Error(
+              `Could not parse required field "${JestPlugin._packageCaptureGroup}" from "packageDir" tag in ` +
+                `"${configurationFilePath}".`
+            );
+          }
+          // Resolve to the root of the package (not the module referenced by the package)
+          const resolvedPackagePath: string = Import.resolvePackage({
+            baseFolderPath: path.dirname(configurationFilePath),
+            packageName
```

Comment:
What if they specify a module path inside a package? You could filter that out via the regexp by only allowing things that look like package names.

Reply:
Going to catch it after running regexp so that a friendly error can be outputted.